src/app/services/bungie-api.service.ts

```diff
@@ -1,571 +1,675 @@
-      components: [
-        DestinyComponentType.ItemCommonData
-      ]
-    })
-
-    let transferResult = false;
-
-    if (!r1) return false;
-    if (r1.Response.characterId != targetCharacter) {
-      if (r1.Response.item.data?.location != 2) {
-        await this.moveItemToVault(r1.Response.item.data?.itemInstanceId || "");
-        r1 = await getItem(d => this.$http(d), {
-          membershipType: destinyMembership.membershipType,
-          destinyMembershipId: destinyMembership.membershipId,
-          itemInstanceId: itemInstanceId,
-          components: [
-            DestinyComponentType.ItemCommonData
-          ]
-        })
-      }
-
-      const payload = {
-        "characterId": targetCharacter,
-        "membershipType": 3,
-        "itemId": r1?.Response.item.data?.itemInstanceId || "",
-        "itemReferenceHash": r1?.Response.item.data?.itemHash || 0,
-        "stackSize": 1,
-        "transferToVault": false
-      }
-
-      transferResult = !!await transferItem(d => this.$httpPost(d), payload);
+    async $httpPost(config: HttpClientConfig) {
+        return this.http
+            .post<any>(config.url, config.body, {
+                params: config.params,
+                headers: {
+                    "X-API-Key": environment.apiKey,
+                    Authorization: "Bearer " + this.authService.accessToken,
+                },
+            })
+            .toPromise()
+            .catch(async (err) => {
+                console.error(err);
+            });
     }
-    if (equip) {
-      let equipPayload = {
-        "characterId": targetCharacter,
-        "membershipType": 3,
-        "stackSize": 1,
-        "itemId": r1?.Response.item.data?.itemInstanceId || "",
-        "itemReferenceHash": r1?.Response.item.data?.itemHash || 0,
-      }
-      transferResult = !!await equipItem(d => this.$httpPost(d), equipPayload)
+
+    async $http(config: HttpClientConfig) {
+        return this.http
+            .get<any>(config.url, {
+                params: config.params,
+                headers: {
+                    "X-API-Key": environment.apiKey,
+                    Authorization: "Bearer " + this.authService.accessToken,
+                },
+            })
+            .toPromise()
+            .catch(async (err) => {
+                console.error(err);
+                if (environment.offlineMode) {
+                    console.debug("Offline mode, ignoring API error");
+                    return;
+                }
+                if (err.error?.ErrorStatus == "SystemDisabled") {
+                    console.info("System is disabled. Revoking auth, must re-login");
+                    await this.authService.logout();
+                }
+                if (err.ErrorStatus != "Internal Server Error") {
+                    console.info("API-Error");
+                    //await this.authService.logout();
+                }
+                // TODO: go to login page
```

Comment:
Issue or obsolete?

Reply:
Should be https://github.com/Mijago/D2ArmorPicker/issues/105